api/controllers/AuthController.ts

```diff
@@ -65,6 +71,15 @@ export class AuthController {
     return { error: null };
   }
 
+  @UseBefore(UserAuthentication)
+  @Post('/emailModification')
+  async emailModification(@Body() emailModificationRequest: EmailModificationRequest,
+    @AuthenticatedUser() user: UserModel): Promise<EmailModificationResponse> {
+    const updatedUser = await this.userAuthService.modifyEmail(user, emailModificationRequest.email);
```

Comment:
nit: we can leave the method name as `modifyEmail`, since we use nouns for REST endpoints but verbs for method names (since the verb in REST is provided by the request type (GET, POST, etc))